lib/src/logic_base.dart

```diff
@@ -121,7 +121,7 @@ class JsonLogic {
 
     // easy syntax for unary operators, like {"var" : "x"} instead of strict {"var" : ["x"]}
     var op = _getOperator(logic);
-    List values = logic[op] is List ? logic[op] : [ logic[op] ];
+    List? values = logic[op] is List ? logic[op] : [ logic[op] ];
```

Comment:
This is nullable due to logic being a Map<dynamic, dynamic> which can contain nullable return types. Thus, the output can be null or [null]. However, the bang operators state that we know that there should be a value. If there is an edge case where there is not a value then a runtime error will be generated.

Reply:
All the other changes are based off this one change.

Reply:
If we removed the `?` operator and the `!` operators from the `apply` function, how could we determine that the function is no longer null safe?

Reply:
Check to see if the list is null? E.g. if (values == null) => logic.